aptos-move/block-executor/src/txn_last_input_output.rs

```diff
@@ -35,31 +36,36 @@ use std::{
 
 type TxnInput<T> = CapturedReads<T, ModuleId, CompiledModule, Module, AptosModuleExtension>;
 
-macro_rules! forward_on_success_or_skip_rest {
-    ($self:ident, $txn_idx:ident, $f:ident) => {{
-        $self.outputs[$txn_idx as usize]
-            .load()
-            .as_ref()
-            .map_or_else(Vec::new, |txn_output| match txn_output.as_ref() {
-                ExecutionStatus::Success(t) | ExecutionStatus::SkipRest(t) => t.$f(),
+macro_rules! with_success_or_skip_rest {
```

Comment:
does this need to be a macro? it seems doable to have a function on self to return Output and chain the operation afterwards?
```
  fn success_or_skip_rest(&self) -> Result<Output> {
                match output.as_ref() {
                ExecutionStatus::Success(t) | ExecutionStatus::SkipRest(t) => t,
                ExecutionStatus::Abort(_)
                | ExecutionStatus::SpeculativeExecutionAbortError(_)
                | ExecutionStatus::DelayedFieldsCodeInvariantError(_) => $fallback,
            }
  }
```